gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -284,19 +297,32 @@ public void execute(Task _task) {
             // Set compression in constructor so that task output has the right name from the start.
             task.setCompression(Compression.GZIP);
             task.getArchiveExtension().set("sls.tgz");
-
-            task.dependsOn(startScripts, initScript, checkScript, yourkitAgent, yourkitLicense);
-            task.dependsOn(copyLauncherBinaries, launchConfigTask, manifest, manifestClassPathTask);
+            task.dependsOn(
+                    startScripts,
+                    initScript,
+                    checkScript,
+                    yourkitAgent,
+                    yourkitLicense,
+                    copyLauncherBinaries,
+                    launchConfigTask,
+                    manifest,
+                    manifestClassPathTask,
+                    javaAgentConfiguration);
         });
 
         project.afterEvaluate(_p -> launchConfigTask.configure(task -> {
+            task.getJavaAgents().setFrom(javaAgentConfiguration);
             if (distributionExtension.getEnableManifestClasspath().get()) {
-                task.setClasspath(manifestClassPathTask.get().getOutputs().getFiles());
+                task.getClasspath().builtBy(manifestClassPathTask.get());
```

Comment:
why do we need to set the classPath to be builtBy?

Reply:
No idea, it seemed like the right thing to do to source the objects, I assumed that helped build a cleaner task dependency graph. I can remove these if that's not what it's for.

Reply:
I think if you use these methods then you can avoid having to set `dependsOn` everywhere as Gradle can work it what it need to run to build these files.